### DISCUSSION TITLE
Pass a class component to ReactRenderer

### DISCUSSION THREAD

**Discussion by iamursky** [ueberdosis/tiptap#1904] [General Questions] [answered]
How to pass a class component to `ReactRenderer`? It says:

```
Argument of type 'typeof SlashCommandsComponent' is not assignable to parameter of type 'ComponentType'.
Property '$$typeof' is missing in type 'typeof SlashCommandsComponent' but required in type
'ForwardRefExoticComponent<{ items: any[]; command: any; } & RefAttributes<unknown>>'.
```

<img width="1016" alt="Screenshot 2021-09-17 at 17 14 11" src="https://user-images.githubusercontent.com/33891482/133798006-f133e942-138f-4053-8b42-42218c1a353d.png">

**Comment by philippkuehn** [comment, marked as answer]
This was a bug and should be fixed with [this patch](https://github.com/ueberdosis/tiptap/commit/fe47c388ce5ef49c286302a7b771a680c2f0c239).

**Reply by iamursky** [reply]
Until then, I used [useImperativeHandle](https://reactjs.org/docs/hooks-reference.html#useimperativehandle) inside functional component to expose the `onKeyDown` method:

```typescript
import { SuggestionKeyDownProps } from "@tiptap/suggestion";
import { forwardRef, useImperativeHandle } from "react";

export const SlashCommandsComponent = forwardRef(
  function SlashCommandsComponentWithRef(props, ref) {
    useImperativeHandle(ref, () => ({
      onKeyDown: ({ event }: SuggestionKeyDownProps) => {},
    }), []);
})
```
